Automattic/jetpack
# Sharing: Deprecate Google+ Sharing Button

See #10950.

#### Changes proposed in this Pull Request:
* Removes from available buttons.
* Adds style and warning tooltip to enabled buttons, preview, and front-end display.
* Removes from front-end display unless `current_user_can( 'manage_options' )`.

#### Copy
The button says:
> Google+ has shut down

The tooltip says:
> The Google+ service has shut down. This sharing button is not displayed to your visitors and should be removed.


#### Screenshots
Settings -> Sharing -> Sharing Buttons:
<img width="493" alt="screen shot 2019-01-24 at 5 47 44 pm" src="https://user-images.githubusercontent.com/125994/51719968-69492180-2000-11e9-9f6b-1d5607b00cd1.png">

Blog Post:
<img width="823" alt="screen shot 2019-01-24 at 5 48 07 pm" src="https://user-images.githubusercontent.com/125994/51719983-7d8d1e80-2000-11e9-9a78-66c60838ff98.png">

#### Testing instructions:
1. `git checkout master`
2. Go to Settings -> Sharing -> Sharing Buttons
3. Add a "Google" button
4. Make "Show buttons on" is checked for at least "Posts".
5. `git checkout deprecate/google-plus-sharing-button`
6. `yarn build`
7. Go to Settings -> Sharing -> Sharing Buttons
8. See that the "Google+" button is dimmed, crossed-out, and has a "has shut down" title/tooltip.
9. In the preview section, see that the "Google+" button says "has shut down", is dimmed, crossed out, and has a "has shut down" title/tooltip.
10. Go to a blog post as a site admin. See the dimmed, etc. Google+ button.
11. Go to a blog post as as a logged out visitor. See no Google+ button.
12. Play around with different "Button style" options both in the admin preview and on the front-end.

#### Proposed changelog entry for your changes:
Deprecate the Google+ Sharing Button.
